code/game/objects/items/explosives/mine.dm

```diff
@@ -63,7 +101,9 @@
 /obj/item/explosive/mine/attack_self(mob/living/user)
 	if(!..())
 		return
-
+	if(needs_digging && user.loc && is_mainship_level(user.z))
```

Comment:
user.loc and is_mainship_level(user.z) are redundant 

if user has a z, it has a loc in form of a turf

```suggestion
	if(needs_digging && is_mainship_level(user.z))
```